src/ClockPanel.tsx

```diff
@@ -1,119 +1,84 @@
 import './external/moment-duration-format';
 import { getTemplateSrv } from '@grafana/runtime';
 
-interface Props extends Themeable2, PanelProps<ClockOptions> {}
-interface State {
-  // eslint-disable-next-line
-  now: Moment;
-}
+interface Props extends PanelProps<ClockOptions> {}
 
 export function getTimeZoneNames(): string[] {
   return (moment as any).tz.names();
 }
 
-class UnthemedClockPanel extends PureComponent<Props, State> {
-  timerID?: any = 0;
-  state = { now: this.getTZ(), timezone: '' };
-
-  componentDidMount() {
-    this.initTimers();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { options, data } = this.props;
-    const { options: prevOptions, data: prevData } = prevProps;
-
-    if (options.refresh !== prevOptions.refresh) {
-      this.initTimers();
-    }
-
-    if (prevData !== data) {
-      this.onPanelRefresh();
-    }
+function getMoment(tz?: string): Moment {
+  if (!tz) {
+    tz = (moment as any).tz.guess();
+  } else {
+    tz = getTemplateSrv().replace(tz);
   }
+  return (moment() as any).tz(tz);
+}
 
-  initTimers = () => {
-    const { refresh } = this.props.options;
-
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
-
-    if (refresh === ClockRefresh.dashboard) {
-      return this.tick();
-    }
-
-    const delay = 1000; // 1sec
-    this.timerID = setInterval(
-      () => this.tick(),
-      delay // 1 second or 1 min
-    );
-  };
-
-  onPanelRefresh = () => {
-    const { refresh } = this.props.options;
-    if (refresh === ClockRefresh.dashboard) {
-      this.tick();
-    }
-  };
-
-  componentWillUnmount() {
-    if (this.timerID) {
-      clearInterval(this.timerID);
-      this.timerID = 0;
-    }
+function getTimeFormat(clockType: ClockType, timeSettings: TimeSettings): string {
+  if (clockType === ClockType.Custom && timeSettings.customFormat) {
+    return timeSettings.customFormat;
   }
 
-  tick() {
-    const { timezone } = this.props.options;
-    this.setState({ now: this.getTZ(timezone) });
+  if (clockType === ClockType.H12) {
+    return 'h:mm:ss A';
   }
 
-  getTimeFormat() {
-    const { clockType, timeSettings } = this.props.options;
-
-    if (clockType === ClockType.Custom && timeSettings.customFormat) {
-      return timeSettings.customFormat;
-    }
+  return 'HH:mm:ss';
+}
 
-    if (clockType === ClockType.H12) {
-      return 'h:mm:ss A';
+export function ClockPanel(props: Props) {
+  const [now, setNow] = useState<Moment>(getMoment(props.options.timezone));
+  const { options, width, height } = props;
+  console.log(options);
+  const theme = useTheme2();
+  const { timezone, dateSettings, timezoneSettings, bgColor } = options;
+
+  const className = css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-flow: column wrap;
+    text-align: center;
+    background-color: ${!bgColor ? theme.colors.background.primary : theme.v1.visualization.getColorByName(bgColor)};
+  `;
```

Comment:
Maybe use the `getStyles` convention to define this class? It will memo it depending on passed parameters.

Example: 
https://github.com/grafana/grafana-plugin-examples/blob/main/examples/panel-datalinks/src/components/DataLinksPanel.tsx#L68

Reply:
I checked the usage of this pattern and I could not find a meaningful way to use it and still have the props available. Because most of these styles are constructed using the passed props. here the bgColor and further down the timezones.

I moved the css classNames to use useMemo so it'll take care of that. Keep in mind this refactor focuses on moving the class component into a functional component and this is an extra optimization on top of that. I'd like this PR to keep focus on refactoring into a function component and not a general make over of the plugin.

Reply:
Ok, but for me it feels like doing smaller improvements like this (boy scout rule) while doing bigger refactorings makes sense. But track it a follow up issue. Perfect as an onboarding task 👍🏻 

Here is an example where you pass parts of props to the `getStyles` pattern:
https://github.com/grafana/grafana/blob/eade49f333e8e12b48fcab9ff3176765fb6c3cf2/public/app/features/canvas/elements/text.tsx#L97